### PR TITLE
Move integration test caching to reusable workflow

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
# Issue

Caching workflow cannot easily be shared between repositories

# Solution

Move workflow to https://github.com/canonical/data-platform
Depends on https://github.com/canonical/data-platform-workflows/pull/1

# Context

#78 

# Testing

Check github actions output

# Release Notes

Move integration test caching to reusable workflow

### CODE REVIEW

**Review by paulomach — APPROVED** [review]
looking nice

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -2,62 +2,69 @@
...
-                # Application charm version does not match mysql charm version
-                # Use latest available version for application charm
-                version = available_versions[-1]
-            return f"local:./{charm_path/charm_name}_ubuntu-{version}-amd64.charm"
+        packed_charms = json.loads(os.environ["CI_PACKED_CHARMS"])
+
+        async def _build_charm(charm_path, bases_index: int = None) -> Path:
+            for charm in packed_charms:
+                if Path(charm_path) == Path(charm["directory_path"]):
+                    if bases_index is None or bases_index == charm["bases_index"]:
```
Comment (shayancanonical) [inline tests/integration/conftest.py:52]: minor nit: the two conditions can be combined with an `and`

**Review by shayancanonical — APPROVED** [review]
This is great! LGTM

It might be nice to create a small document that outlines exactly what one needs to do to adopt the caching in their repository

**Review by paulomach — APPROVED** [review]
let's go!

File: tests/integration/conftest.py (hunk 1 above)
Comment (shayancanonical) [reply tests/integration/conftest.py:52]: suggestion: we could put () around the second statement

**Review by taurus-forever — APPROVED** [review]
I like it! (c)

**Pull request merged**